text/trace/0192-messaging-semantic-conventions-spec.md

```diff
@@ -0,0 +1,659 @@
+Kafka.
+
+This attributes is required for producer spans modelling `publish` or `create`
+operations. It is optional for consumer spans. The name of the source a message
+is received from can be different from the name of the target a message is sent
+to. If this attribute is used on consumer spans, it should be set to the name
+of the target the message was initially published to by the producer.
+
+##### `messaging.destination.template`
+
+In some instances, message destination names are constructed from templates. An
+example would be a destination name involving a user name or product id.
+Although the destination name in this case is of high cardinality, the
+underlying template is of low cardinality and can be effectively used for
+grouping and searching spans.
+
+This attribute is optional, but recommended if the destination name is created
+based on such a template.
+
+##### `messaging.destination.kind`
+
+Different brokers have different concepts of message destinations, the most
+popular being queues and topics. One of the most important differences in
+destination kinds is how messages are settled: messages are settled
+individually in queues, whereas messages are settled based on checkpoints in
+topics. Individual brokers might specify additional destination kinds.
+
+##### `messaging.destination.temporary`
+
+If set to `true`, this flag denotes that the destination is a temporary
+destination and might not exist anymore after messages are processed.
+
+##### `messaging.destination.anonymous`
+
+If set to `true`, this flag denotes that the destination is an anonymous
+destination. Anonymous destinations are usually established just for a
+particular set of producers and consumer. Often such destinations are unnamed
+or have an auto-generated name.
+
+##### `messaging.source.name`
+
+The source name defines name of the source of a message, as it is
+specified by the consumer. There are different kinds of targets, varying
+between different message brokers, e. g. queues in RabbitMQ, or topics in
+Kafka.
+
+This attributes is required for consumer spans modelling `deliver`, `receive`,
+or `settle` operations. The name of the source a message is received from can
+be different from the name of the destination a message was sent to.
+
+##### `messaging.source.template`
+
+See [`messaging.destination.template`](#messagingdestinationtemplate).
+
+##### `messaging.source.kind`
+
+See [`messaging.destination.kind`](#messagingdestinationkind).
+
+##### `messaging.source.temporary`
+
+See [`messaging.destination.temporary`](#messagingdestinationtemporary).
+
+##### `messaging.source.anonymous`
+
+See [`messaging.destination.anonymous`](#messagingdestinationanonymous).
+
+##### `net.app.protocol.name`
+
+The name of the underlying protocol which is used to publish and receive
+messages. This should specify application layer protocols (e. g. AMQP, MQTT, or
+HTTP) and not transport or network layer protocols (e. g. TCP, UDP, IP).
```

Comment:
See below https://github.com/open-telemetry/oteps/pull/192#discussion_r929893987

Reply:
Tbh I followed the PR but I still see the `net.app.protocol.name` and version ... even if comment #192 says ....

> "net." is not not the network-layer stuff anymore

I do still think that the usage we are doing of this field with AMQP, MQTT, .. values is something related to the messaging so still using `messaging.` prefix makes more sense than `net.`
What am I missing?

Reply:
@ppatierno [`net.*` are generic attributes](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/span-general.md) used by all hotel specs: HTTP, database, and messaging. 
Reusing attributes across specs makes it simpler for users to learn it once and write queries across different systems.

You can do messaging over HTTP/gRPC (some messaging systems in Azure, AWS, and GCP do it), so use Redis or DB feeds like messaging, so having messaging-specific protocols only will not work for all cases, and duplicating general protocols in messaging spec is not a great solution.

Reply:
@lmolkova so the comment I read "net." is not not the network-layer stuff anymore" is not really true, because on your link I see "Network transport attributes" table. In this case I can understand that we want to share attributes across systems so `net.app.protocol.*` it's related to the application layer protocol and it makes sense.

Reply:
(Kinda OT:)
@ppatierno
> so the comment I read "net. is not not the network-layer stuff anymore" is not really true

I meant network layer, as opposed to application layer. Of course it is still networking-related, but not necessarily the TCP/UDP-level layer of the networking stack.